Actividad 4 Taller Pandas - Numpy 
Instrucciones:

- Formula 1 pregunta de análisis sobre tu conjunto de datos
- Utilizando Pandas y Numpy:
- Limpieza de datos (Pandas: dropna, fillna, replace, etc.)
- Filtrado y selección de datos
- Transformación de columnas, creación de variables nuevas
- Agrupaciones y resúmenes (groupby)
- Conversión a array NumPy para operaciones matemáticas rápidas (media, desvío, normalización, operaciones vectorizadas)
- Responde la pregunta planteada con una breve interpretación.

In [10]:
import pandas as pd
import numpy as np

df = pd.read_csv('imdb_top_movies.csv')
df.head()

,Rank,Title,Year,Rating,Duration,Certificate,Genres,Description,Image URL,Movie URL
0,1,The Shawshank Redemption,1994,9.3,2h 22m,R,Drama,A banker convicted of uxoricide forms a friend...,https://m.media-amazon.com/images/M/MV5BMDAyY2...,https://www.imdb.com/title/tt0111161/
1,2,The Godfather,1972,9.2,2h 55m,R,"Crime, Drama",The aging patriarch of an organized crime dyna...,https://m.media-amazon.com/images/M/MV5BNGEwYj...,https://www.imdb.com/title/tt0068646/
2,3,The Dark Knight,2008,9.1,2h 32m,PG-13,"Action, Crime, Drama",When a menace known as the Joker wreaks havoc ...,https://m.media-amazon.com/images/M/MV5BMTMxNT...,https://www.imdb.com/title/tt0468569/
3,4,The Godfather Part II,1974,9.0,3h 22m,R,"Crime, Drama",The early life and career of Vito Corleone in ...,https://m.media-amazon.com/images/M/MV5BMDIxMz...,https://www.imdb.com/title/tt0071562/
4,5,12 Angry Men,1957,9.0,1h 36m,Approved,"Crime, Drama",The jury in a New York City murder trial is fr...,https://m.media-amazon.com/images/M/MV5BYjE4Nz...,https://www.imdb.com/title/tt0050083/


Pregunta de análisis

¿Cuál es la duración promedio de las películas por década y qué década tiene las películas más largas?

In [11]:
# Limpieza de datos
def convert_duration(d):
    try:
        h, m = 0, 0
        if "h" in d:
            h = int(d.split("h")[0])
            m = int(d.split("h")[1].replace("m", "").strip())
        return h*60 + m
    except:
        return np.nan

df["Duration_min"] = df["Duration"].apply(convert_duration)
df["Year"] = pd.to_numeric(df["Year"], errors="coerce")


In [12]:
# Eliminar filas con datos esenciales faltantes
df = df.dropna(subset=["Duration_min", "Year"])

In [13]:
# Crear columna Década
df["Decade"] = (df["Year"] // 10) * 10

In [14]:
# Agrupación: duración promedio por década
decade_avg = df.groupby("Decade")["Duration_min"].mean().reset_index()

In [15]:
# NumPy: array de duraciones
duration_array = df["Duration_min"].values

duration_mean = np.mean(duration_array)
duration_std = np.std(duration_array)

In [16]:
# Normalización tipo z-score
duration_normalized = (duration_array - duration_mean) / duration_std

decade_avg.head(), duration_mean, duration_std

(   Decade  Duration_min
 0    1920     84.666667
 1    1930    123.666667
 2    1940    118.181818
 3    1950    126.333333
 4    1960    146.375000,
 131.728,
 36.21753740938221)

¿Cuál es la duración promedio de las películas por década y qué década tiene las películas más largas?

Las películas tienden a aumentar su duración a medida que avanzan las décadas.

De acuerdo con los resultados:

La década con las películas más largas es la de 1960, con un promedio de 146.37 minutos.
Esto quiere decir que en los años 60 hubo una tendencia hacia películas más extensas, posiblemente debido a cambios creativos y narrativos en la industria cinematográfica de esa época.